### ISSUE TITLE
terminal.integrated.shell.windows setting Doesn't support environment variables

### ISSUE THREAD

**Issue by richardspence** [Microsoft/vscode#7344]
- VSCode Version: 1.2.0
- OS Version: Win 10 enterprise preview

Steps to Reproduce:
1.  Set user setting to  "terminal.integrated.shell.windows": "%programfiles%\Git\git-bash.exe"
2.  Open terminal ctrl+`

Expected:
Terminal opens.

Actual:
Terminal window flickers and closes (presumably crashes).

Workaround:
Use explicit pathing like "C:\Program Files\Git\git-bash.exe" instead.

**Comment by Tyriar** [comment]
@joaomoreno what are your thoughts on adding this? My main concern is consistency with other settings (it would be unclear when environment variables evaluated).

**Comment by joaomoreno** [comment]
Yep, this is a bit strange. Also, since settings can roam, there would need to be a format across all platforms and `%var%` is windows specific. Not a big fan, I'd let this be as designed.

**Comment by Tyriar** [comment]
Closing as designed.

**Comment by ebugusey** [comment]
> Yep, this is a bit strange. Also, since settings can roam, there would need to be a format across all platforms and %var% is windows specific. Not a big fan, I'd let this be as designed.

But `terminal.external.windowsExec` does support this kind of variables, so why `terminal.integrated.shell.windows` isn't? They are windows specific variables, so why shouldn't they support env variables in windows format?

**Comment by Tyriar** [comment]
Duplicate https://github.com/Microsoft/vscode/issues/14973

**Comment by mihailik** [comment]
@Tyriar this is not a duplicate of #149773 `Set environment variable before starting a terminal` (which isresolved).

It is a duplicate of #2809 `Support environment variables when resolving values in settings` (currently not implemented).
